File: go/hamming/hamming.go
package hamming

import (
	"errors"
	"math"
)

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return 0, errors.New("invalid length")
    }
    count := 0
    index := 0
    for index < int(math.Min(float64(len(a)), float64(len(b)))) {
        if a[index] != b[index] {
            count += 1
        }
        index += 1
    }
    return count, nil
}

File: c/hamming/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs) {
  if (!lhs || !rhs) {
    return -1;
  }
  int count = 0;
  int i;
  for (i = 0; lhs[i] != '\0' && rhs[i] != '\0'; ++i) {
    if (lhs[i] != rhs[i]) {
      ++count;
    }
  }
  return lhs[i] != '\0' || rhs[i] != '\0' ? -1 : count;
}
